source/constructs/api/data_source/service.py

```diff
@@ -1624,7 +1637,7 @@
                                   region_name=region_aws
                                   )
     """ :type : pyboto3.secretsmanager """
-    response = secretsmanager.list_secrets()
+    response = secretsmanager.list_secrets(MaxResults=100)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_secrets` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)